recipes/libhal/all/conanfile.py

```diff
@@ -36,14 +36,26 @@ def _compilers_minimum_version(self):
             "apple-clang": "14"
         }
 
+    @property
+    def _bare_metal(self):
+        return self.settings.os == "baremetal"
+
     def export_sources(self):
         export_conandata_patches(self)
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.83.0", transitive_headers=True)
+        # NOTE: from the author, kammce, to the conan team. boost-leaf is tied
+        # to the 2.0.1 and must be kept for it to work correctly.
+        version = Version(self.version)
+        if version == "2.0.1":
+            self.requires("boost-leaf/1.81.0",)
```

Comment:
The `boost-leaf` package is deprecated and should be using `boost` instead. The version 1.83.0 (and older) has leaf integrated: https://www.boost.org/doc/libs/1_83_0/libs/leaf/doc/html/index.html Is it not enough?

Reply:
No because packages 2.0.1 and below use the single header file in its includes. Boost does not include the single header file. So to keep the software the same and building, we must use the deprecated Boost.LEAF package. Or opt to remove the 2.0.1 package from the index. Or changing 2.0.1 to accommodate the boost header only. I think using boost leaf is the better alternative here.

Reply:
Unfortunately, using boost-leaf package is not an option, it's deprecated already. We just didn't remove it due CCI policy to avoid breaking users. 

Let's drop 2.0.1 then, it will consume less maintenance and will "enforce" users to use the latest version of libhal instead.

Reply:
I thought that was against CCI guidelines to drop packages? I'm all for it though.

Reply:
@kammce we actually don't remove the package from Conan Center, but the version from conandata.yml, so we no longer build new revisions: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/sources_and_patches.md#removing-old-versions

Reply:
@kammce after talking with @RubenRBS, is clear supporting boost-leaf for your recipe is the best case, I didn't understand that libhal was not designed to support Boost itself. So, boost-leaf is a risk that we can assume, but need to be clear for users, with a `self.output.warning` message. Rubén should talk to you via Slack to align better. Thank you for bringing more details about your case.